pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -0,0 +1,286 @@
+package buildplan
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	model "github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+	"github.com/go-openapi/strfmt"
+)
+
+// NewMapFromBuildPlan creates an artifact map from a build plan. It creates a
+// lookup table and calls the recursive function buildMap to build up the
+// artifact map by traversing the build plan from the terminal targets through
+// all of the runtime dependencies for each of the artifacts in the DAG.
+func NewMapFromBuildPlan(build *model.Build) (artifact.Map, error) {
+	if build == nil {
+		// The build plan can be nil when calculating the changeset for a build
+		// that has not been activated yet.
+		return nil, nil
+	}
```

Comment:
This feels like a check that should've happened before this function was called.